Src/AmrCore/AMReX_AmrParticles.H

```diff
@@ -249,40 +249,47 @@ ParticleToMesh (PC const& pc, const Vector<MultiFab*>& mf,
     }
 }
 
-template <int NStructReal, int NStructInt=0, int NArrayReal=0, int NArrayInt=0,
+template <typename T_ParticleType, int NArrayReal=0, int NArrayInt=0,
           template<class> class Allocator=DefaultAllocator>
-class AmrParticleContainer // NOLINT
-    : public ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
+class AmrParticleContainer_impl
+    : public ParticleContainer_impl<T_ParticleType, NArrayReal, NArrayInt, Allocator>
 {
 
 public:
 
-    typedef Particle<NStructReal, NStructInt> ParticleType;
+    using ParticleType = T_ParticleType;
 
-    AmrParticleContainer () = default;
+    AmrParticleContainer_impl ()
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>()
+    {
+    }
 
-    AmrParticleContainer (AmrCore* amr_core)
-        : ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>(amr_core->GetParGDB())
+    AmrParticleContainer_impl (AmrCore* amr_core)
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>(amr_core->GetParGDB())
     {
     }
 
-    AmrParticleContainer (const Vector<Geometry>            & geom,
-                          const Vector<DistributionMapping> & dmap,
-                          const Vector<BoxArray>            & ba,
-                          const Vector<int>                 & rr)
-        : ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>(geom, dmap, ba, rr)
+    AmrParticleContainer_impl (const Vector<Geometry>            & geom,
+                               const Vector<DistributionMapping> & dmap,
+                               const Vector<BoxArray>            & ba,
+                               const Vector<int>                 & rr)
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>(geom, dmap, ba, rr)
     {
     }
 
-    ~AmrParticleContainer () override = default;
+    virtual ~AmrParticleContainer_impl () = default;
```

Comment:
Either back to:
```suggestion
    ~AmrParticleContainer_impl () override = default;
```
or, since `ParticleContainer_impl` is the base class, no virtual/override at all?